pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
```

Comment:
Create a test that calls validation with an object and verifies it doesn't
change, but do so with a range of input.  Fuzzing would allow you to test
many more outcomes, but I'd be ok with a test that from inspection is the
same.  I acknowledge Jordan's concerns, but deep copy is a hammer and I'm
not concerned with mutation if we have a very focused test.

On Wed, Jun 8, 2016 at 11:07 AM, Gabe Montero <notifications@github.com>
wrote:

> In pkg/deploy/api/validation/validation.go
> <https://github.com/openshift/origin/pull/9167#discussion_r66274074>:
>
> > @@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
> >  	if spec.Template == nil {
> >  		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
> >  	} else {
> > +		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
>
> Apologies, don't know what is meant by "fuzz test". Can you clarify
> @smarterclayton <https://github.com/smarterclayton> ?
>
> Pending that, I'll move forward with removing the deep copy and
> incorporating a temp update / revert pattern.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9167/files/7108af0c910f3080a151b4fbd222c31265396ccb#r66274074>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p4UxNFf_r1BDUEyYtrNMFzbI6pWyks5qJtq0gaJpZM4It4tM>
> .
>